catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStorePublisher.java

```diff
@@ -88,79 +114,69 @@ public void setUnavailable() {
+            RegistryStore registryStore = bundleContext.getService(reference);
 
             registryStoreMap.remove(registryStore.getConfigurationPid());
         }
     }
 
-    void registryPublish(RegistryStore registryStore) {
+    void registryPublish(RegistryStore registryStore, boolean publish) {
 
         if (registryStore.getRegistryId()
                 .isEmpty()) {
-            return;
-        }
-        try {
-
-            Optional<Metacard> registryIdentityMetacardOpt =
-                    Security.runAsAdminWithException(() -> federationAdminService.getLocalRegistryIdentityMetacard());
-
-            if (registryIdentityMetacardOpt.isPresent()) {
-                Metacard registryIdentityMetacard = registryIdentityMetacardOpt.get();
-                String localRegistryId = registryIdentityMetacard.getAttribute(
-                        RegistryObjectMetacardType.REGISTRY_ID)
-                        .getValue()
-                        .toString();
-                Security.runAsAdminWithException(() -> {
-                    registryPublicationService.publish(localRegistryId,
-                            registryStore.getRegistryId());
-                    return null;
-                });
+            StringBuilder logMessage = new StringBuilder(
+                    "RegistryStore missing id. Unable to complete ");
+            if (!publish) {
+                logMessage.append("un");
             }
-        } catch (Exception e) {
-            LOGGER.error("Failed to publish registry configuration to {}",
-                    ((RegistryStoreImpl) registryStore).getRemoteName());
-        }
-    }
-
-    void registryUnpublish(RegistryStore registryStore) {
-
-        if (registryStore.getRegistryId()
-                .isEmpty()) {
+            logMessage.append("publish request.");
+            LOGGER.warn(logMessage.toString());
             return;
         }
-        try {
 
-            Optional<Metacard> registryIdentityMetacardOpt =
-                    Security.runAsAdminWithException(() -> federationAdminService.getLocalRegistryIdentityMetacard());
-
-            if (registryIdentityMetacardOpt.isPresent()) {
-                Metacard registryIdentityMetacard = registryIdentityMetacardOpt.get();
-                String localRegistryId = registryIdentityMetacard.getAttribute(
-                        RegistryObjectMetacardType.REGISTRY_ID)
-                        .getValue()
-                        .toString();
-                Security.runAsAdminWithException(() -> {
-                    registryPublicationService.unpublish(localRegistryId,
-                            registryStore.getRegistryId());
-                    return null;
-                });
-            }
-        } catch (Exception e) {
-            LOGGER.error("Failed to unpublish registry configuration to ",
-                    ((RegistryStoreImpl) registryStore).getRemoteName());
-        }
+        executor.schedule(() -> {
```

Comment:
Since this is happening on a single thread, there is a chance the publish call may be somewhat slow and hold up execution or possible faster than the catalogStore, thus fail to publish. Separating this into a separate thread and then delaying a few seconds greatly reduces these issues.